streams/src/main/java/org/apache/kafka/streams/processor/StreamPartitioner.java

```diff
@@ -58,5 +62,24 @@
      * @param numPartitions the total number of partitions
      * @return an integer between 0 and {@code numPartitions-1}, or {@code null} if the default partitioning logic should be used
      */
+    @Deprecated
     Integer partition(String topic, K key, V value, int numPartitions);
+
+    /**
+     * Determine the partition numbers to which a record, with the given key and value and the current number
+     * of partitions, should be multi-casted to.
```

Comment:
nit: wording here is a bit awkward, sounds like we're talking about the current number of partitions of the record, not the topic -- maybe something like this?
```suggestion
     * Determine the number(s) of the partition(s) to which a record with the given key and value should be sent, 
     * for the given topic and current partition count
```